test/unit/communication/test_broadcast_tile.cpp

```diff
@@ -12,44 +12,42 @@
 
 #include <gtest/gtest.h>
 
-#include "internal/helper_communicators.h"
+#include "dlaf_test/helper_communicators.h"
+#include "dlaf_test/util_tile.h"
 #include "dlaf_test/util_types.h"
 
 #include "dlaf/tile.h"
-#include "dlaf/util_tile.h"
 
-using BroadcastTileTest = dlaf_test::SplittedCommunicatorsTest;
+using namespace dlaf;
+using namespace dlaf_test;
+using namespace dlaf::comm;
+
+using BroadcastTileTest = SplittedCommunicatorsTest;
 
 TEST_F(BroadcastTileTest, SyncTile) {
   using TypeParam = std::complex<float>;
 
   TypeParam data[4];
-  dlaf::Tile<TypeParam, dlaf::Device::CPU> tile({2, 2}, {data, 4}, 2);
+  Tile<TypeParam, dlaf::Device::CPU> tile({2, 2}, {data, 4}, 2);
 
   EXPECT_EQ(2, tile.size().rows());
   EXPECT_EQ(2, tile.size().cols());
 
-  auto message_values = [](dlaf::TileElementIndex index) -> TypeParam {
-    return dlaf_test::TypeUtilities<TypeParam>::element((index.row() + 1) + (index.col() + 1) * 10,
-                                                        (index.col() + 1));
+  auto message_values = [](const TileElementIndex& index) -> TypeParam {
+    return TypeUtilities<TypeParam>::element((index.row() + 1) + (index.col() + 1) * 10,
+                                             (index.col() + 1));
   };
 
-  if (isMasterInSplitted()) {
-    for (int j = 0; j < tile.size().cols(); ++j)
-      for (int i = 0; i < tile.size().rows(); ++i)
-        *tile.ptr({i, j}) = message_values({i, j});
+  if (splitted_comm.rank() == 0) {
+    tile_test::set(tile, message_values);
 
-    dlaf::comm::broadcast::send(dlaf::comm::make_message(tile), splitted_comm);
+    broadcast::send(make_message(tile), splitted_comm);
   }
   else {
-    for (int j = 0; j < tile.size().cols(); ++j)
-      for (int i = 0; i < tile.size().rows(); ++i)
-        EXPECT_NE(message_values({i, j}), *tile.ptr({i, j}));
+    CHECK_TILE_NE(message_values, tile);
```

Comment:
Avoid checks on unitialized memory. It is not guaranteed that the memory of the tile is different from the message value.
Prefer the approach which sets the elements (e.g. set to NaN or also 0 since all your elements (real and imaginary parts for complex) are >0).